crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -224,6 +224,8 @@ impl<'db> InnerIntersectionBuilder<'db> {
                 self.positive.retain(|elem| !neg.contains(elem));
                 self.negative.retain(|elem| !pos.contains(elem));
             }
+            // Note: `[Type::Todo]` is included to show that an unimplemented type was added, but
+            // this can lead to cases where separate `Type::Todo` cancel eachother
```

Comment:
The correct behavior is for Todo to not cancel itself in an intersection, same as Any and Unknown.

I would not add this comment, here or in `add_negative`; rather, we have an existing TODO comment in `add_negative` to stop self-cancelling Any and Unknown; let's just mention Todo in that comment also (and maybe add the same comment here in `add_positive`)